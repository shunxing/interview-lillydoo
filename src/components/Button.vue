<template>
  <div class="button">{{ label }}</div>
</template>

<script>
export default {
  name: "Button",
  props: { label: String },
};
</script>

<style lang="scss" scoped>
.button {
  height: 30px;
  border: 1px black solid;
  max-width: 30px;
  min-width: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 5px;
}
</style>
