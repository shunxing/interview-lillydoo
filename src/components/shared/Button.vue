<template>
  <div v-bind:class="buttonClass">
    <span>{{ label }}</span
    ><span>{{ secondaryLabel }}</span>
  </div>
</template>

<script>
export default {
  name: "Button",
  props: { label: String, isSelected: Boolean, secondaryLabel: String },
  computed: {
    buttonClass: function () {
      return {
        button: true,
        ["button--active"]: this.isSelected,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.button {
  min-height: 30px;
  border: 1px $disabled-color solid;
  max-width: 30px;
  min-width: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 10px 5px;
  color: $disabled-color;
  display: flex;
  flex-direction: column;
  align-items: center;

  &--active {
    background: #00afab;
    color: white;
    border-color: transparent;
  }
}
</style>
