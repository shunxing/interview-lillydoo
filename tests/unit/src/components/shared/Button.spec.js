import { shallowMount, mount } from "@vue/test-utils";
import Button from "@/components/shared/Button.vue";

describe("Button.vue", () => {
  it("should render a button correctly with no specific props", () => {
    const wrapper = mount(Button);
    expect(wrapper.find(".button").exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render label props when passed", () => {
    const label = "testButtonLabel";
    const wrapper = shallowMount(Button, { props: { label } });
    expect(wrapper.text()).toMatch(label);
  });

  describe("class", () => {
    it("should render active when selected", () => {
      const wrapper = shallowMount(Button, { props: { isSelected: false } });
      expect(wrapper.find(".button").classes()).not.toContain("button--active");
      expect(wrapper.find(".button").classes()).toContain("button");
    });

    it("should render active when selected", () => {
      const wrapper = shallowMount(Button, { props: { isSelected: true } });
      expect(wrapper.find(".button").classes()).toContain("button--active");
      expect(wrapper.find(".button").classes()).toContain("button");
    });
  });
});
