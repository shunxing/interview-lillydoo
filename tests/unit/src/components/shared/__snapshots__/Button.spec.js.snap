// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue should render a button correctly with no specific props 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {
        "0": Object {},
      },
      "__props": Object {
        "0": Array [
          Object {},
          Array [],
        ],
      },
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
      id="app"
    >
      <div
        class="button"
      >
        <span
          class="button__label"
        />
        <span
          class="button__secondary-label"
        />
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {},
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
        Object {
          "__emits": Object {
            "0": null,
          },
          "__props": Object {
            "0": Array [],
          },
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.0-rc.10",
  },
  "__isFunctionalComponent": false,
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
    "buttonClass": Object {
      "button": true,
      "button--active": false,
      "button--inverted": false,
      "undefined": undefined,
    },
    "class": undefined,
    "inverted": false,
    "isSelected": false,
    "label": undefined,
    "secondaryLabel": undefined,
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
